### PR TITLE
Various fixes + some support for using the web API

### PR DESCRIPTION
**Pull request by asayers**
Testing performed:

```
$ SLACK_API_TOKEN=[...] git rebase -ix "stack test --pedantic" master
[no errors, warnings, or test failures]
```

### CODE REVIEW

**Comment by asayers** [comment]
This branch doesn't change the interface of `Web.Slack`, so think it's fine to include in a minor release.

**Comment by asayers** [comment]
When reviewing, it's probably best to ignore b34ac6d ("Alphabeticize"):
```
$ git revert -n -Xtheirs b34ac6d
$ git diff master # etc.
```

File: src/Web/Slack/Types/Item.hs [hunk 1]
```diff
@@ -18,15 +18,22 @@ data Item = MessageItem ChannelId MessageUpdate
           | FileCommentItem File Comment
           | ChannelItem ChannelId
           | IMItem ChannelId
-          | GroupItem ChannelId deriving Show
+          | GroupItem ChannelId
+          | EmbeddedMessageItem ChannelId SlackTimeStamp
```
Comment (mpickering) [inline src/Web/Slack/Types/Item.hs]: Should these embedded items be a new datatype?

**Comment by mpickering** [comment]
Looks generally good. See the one comment.

File: src/Web/Slack/Types/Item.hs (hunk 1 above)
Comment (asayers) [inline src/Web/Slack/Types/Item.hs]: Yes, I think you're right.

**Pull request merged**